### ISSUE TITLE
Decoding quality is very poor in some situation.

### ISSUE THREAD

**Issue by tyan0** [cisco/openh264#3712]
For example, decoding [original.mp4](https://github.com/cisco/openh264/assets/32232575/b6a457c9-6498-4aa2-b854-85c546389c04) by libopenh264 and encoding again by libx264 results in:

https://github.com/cisco/openh264/assets/32232575/8b2f36db-d92a-4df9-a927-210338b9e076

while decoding the same video by h264 decoder in ffmpeg and encoding again by libx264 results in:

https://github.com/cisco/openh264/assets/32232575/9c9de0d2-568e-4ae2-ac94-909d9e8af46a

Is any important feature of h264 missing from openh264?

**Comment by joakim-tjernlund** [comment]
Still poor quality with your latest fixes waiting fro review?

**Comment by lesueurnicolas27** [comment]
[README.md](https://github.com/cisco/openh264/files/13719489/README.md)
[README.md](https://github.com/cisco/openh264/files/13719493/README.md)

**Comment by tyan0** [comment]
> Still poor quality with your latest fixes waiting fro review?

Yes. The pull request https://github.com/cisco/openh264/pull/3707 fixes only the reordering of the decoded frames. I guess this issue is in the decoding algorithm.

**Comment by joakim-tjernlund** [comment]
Just tested viewing this mp4 in firefox and it looks better but some artifacts are still there
